python-package/xgboost/spark/data.py

```diff
@@ -0,0 +1,262 @@
+import os
+from typing import Iterator
+import numpy as np
+import pandas as pd
+from scipy.sparse import csr_matrix
+from xgboost import DMatrix
+
+
+# Since sklearn's SVM converter doesn't address weights, this one does address weights:
+def _dump_libsvm(features, labels, weights=None, external_storage_precision=5):
```

Comment:
I'm not entirely sure why do you need to dump data into libsvm. If it's about external memory, we should go with data iterator instead.

Reply:
> dump data into libsvm

this is an option (turn off by default), this approach can reduce memory. If there's data iterator API, we don't need this.

> go with data iterator

In current code we don't use DMatrix data iterator (we didn't find related API ). Does xgboost python support DMatrix data iterator API now ?

Reply:
It supports a form of custom iterator for external memory: https://xgboost.readthedocs.io/en/stable/tutorials/external_memory.html#data-iterator . I will work on the `DeviceQuatnileDMatrix` support after this initial merge, which uses a very similar interface. But for the first PR, I don't think external memory support is necessary.